app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SplashActivity.kt

```diff
@@ -0,0 +1,78 @@
+package com.leoart.koreanphrasebook.ui.splash
+
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import com.badoo.mobile.util.WeakHandler
+import com.leoart.koreanphrasebook.R
+import com.leoart.koreanphrasebook.ui.MainActivity
+import com.leoart.koreanphrasebook.ui.ViewModelFactory
+import com.leoart.koreanphrasebook.ui.sync.SyncActivity
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+class SplashActivity : AppCompatActivity() {
+
+    private val SPLASH_DELAY: Long = 1500
+    private var handler: WeakHandler? = null
+    private var runnable: Runnable? = null
+    private lateinit var model: SplashViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_splash)
+        model = ViewModelProviders.of(
+                this,
+                ViewModelFactory(applicationContext)
+        ).get(SplashViewModel::class.java)
+        model.openNextScreen()
```

Comment:
@greenhorn31 this is wrong ViewModel implementation. You should subscribe to data change, not to request response. Please check https://developer.android.com/topic/libraries/architecture/viewmodel for more information.